packages/arb-token-bridge-ui/src/hooks/useOftTransactionHistory.ts

```diff
@@ -0,0 +1,352 @@
+      optionalDVNThreshold: number
+      requiredDVNs: string[]
+      requiredDVNNames: string[]
+      optionalDVNs: string[]
+      optionalDVNNames: string[]
+    }
+    outboundConfig: {
+      confirmations: number
+      requiredDVNCount: number
+      optionalDVNCount: number
+      optionalDVNThreshold: number
+      requiredDVNs: string[]
+      requiredDVNNames: string[]
+      optionalDVNs: string[]
+      optionalDVNNames: string[]
+      executor: string
+    }
+    ulnSendVersion: string
+    ulnReceiveVersion: string
+  }
+  status: {
+    name: string
+    message: string
+  }
+  created: string
+  updated: string
+}
+
+const getOftTransactionStatus = (message: LayerZeroMessage) => {
+  switch (message.status.name) {
+    case LayerZeroMessageStatus.INFLIGHT ||
+      LayerZeroMessageStatus.CONFIRMING ||
+      LayerZeroMessageStatus.PAYLOAD_STORED:
+      return 'pending'
+    case LayerZeroMessageStatus.DELIVERED:
+      return 'success'
+    case LayerZeroMessageStatus.FAILED || LayerZeroMessageStatus.BLOCKED:
+      return 'failed'
+    default:
+      return 'pending'
+  }
+}
+
+function validateSourceAndDestinationChainIds(message: LayerZeroMessage) {
+  const sourceChainId = getChainIdFromEid(message.pathway.srcEid)
+  const destinationChainId = getChainIdFromEid(message.pathway.dstEid)
+
+  if (!sourceChainId || !destinationChainId) {
+    return false
+  }
+
+  return true
+}
+
+function mapLayerZeroMessageToMergedTransaction(
+  message: LayerZeroMessage
+): LayerZeroTransaction {
+  const sourceChainId = getChainIdFromEid(message.pathway.srcEid)
+  const destinationChainId = getChainIdFromEid(message.pathway.dstEid)
+
+  if (!sourceChainId || !destinationChainId) {
+    throw new Error('Invalid chain ids')
+  }
+
+  const isDeposit = isDepositMode({
+    sourceChainId,
+    destinationChainId
+  })
+
+  return {
+    isOft: true,
+    isCctp: false,
+    sender: message.source.tx.from,
+    direction: isDeposit ? 'deposit' : 'withdraw',
+    status: getOftTransactionStatus(message),
+    createdAt: new Date(message.created).getTime(),
+    resolvedAt:
+      getOftTransactionStatus(message) === 'pending'
+        ? null
+        : new Date(message.updated).getTime(),
+    txId: message.source.tx.txHash,
+    assetType: AssetType.ERC20,
+    uniqueId: null,
+    isWithdrawal: false,
+    blockNum: null,
+    childChainId: isDeposit ? destinationChainId : sourceChainId,
+    parentChainId: isDeposit ? sourceChainId : destinationChainId,
+    sourceChainId,
+    destinationChainId
+  }
+}
+
+export async function updateAdditionalLayerZeroData(
+  tx: LayerZeroTransaction
+): Promise<MergedTransaction> {
+  const { txId } = tx
+  const updatedTx = { ...tx }
+
+  const sourceChainProvider = getProviderForChainId(tx.sourceChainId)
+
+  // extract destination address
+  const sourceChainTx = await sourceChainProvider.getTransaction(txId)
+  const inputDataInterface = new ethers.utils.Interface([
```

Comment:
better add a comment above to link to the ABI source?

Reply:
Added